### PR TITLE
docs: Add more context and links to vault docs

### PR DESCRIPTION
**Pull request by markgoddard**

### CODE REVIEW

File: doc/source/configuration/vault.rst [hunk 1]
```diff
@@ -6,6 +6,18 @@ This document describes how to deploy Hashicorp Vault for
 internal PKI purposes using the
 `StackHPC Hashicorp collection <https://galaxy.ansible.com/stackhpc/hashicorp>`_
 
+Vault may be used as a Certificate Authority to generate certificates for:
+
+* OpenStack internal API
+* OpenStack backend APIs
+* RabbitMQ
+
+TLS support is described in the :kolla-ansible-doc:`Kolla Ansible documentation
```
Comment (Alex-Welsh) [inline doc/source/configuration/vault.rst:15]: I can see that magic links like this work but how? Where is the `kolla-ansible-doc` function defined?

**Review by Alex-Welsh — APPROVED** [review]

File: doc/source/configuration/vault.rst (hunk 1 above)
Comment (markgoddard) [reply doc/source/configuration/vault.rst:15]: See the extlinks stuff in doc/source/conf.py

Comment (markgoddard) [reply doc/source/configuration/vault.rst:15]: and https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html

Comment (Alex-Welsh) [reply doc/source/configuration/vault.rst:15]: Ah, now I can see why I couldn't grep for it. Cool stuff

**Pull request merged**